### ISSUE TITLE
pfsense netdata service

### ISSUE THREAD

**Issue by lordcantide** [firehol/netdata#3079]
On pfsense, the FreeBSD daemon configuration references a path to a netdata PID that is an empty file.
I was able to resolve the issue by changing the pidfile value in file **/user/local/etc/rc.d/netdata** from `pidfile="/var/db/netdata/${name}.pid"` to `pidfile="/var/run/${name}.pid"`
I'm a little dumb on who can actually fix the package for all BSD builds so I figured this would be a good place to log the issue.

**Comment by ktsaou** [comment]
Thanks!

@mmokhi could you please check this. On FreeBSD/pfsense the pid file path is wrong.

**Comment by ktsaou** [comment]
> @ktsaou is there standard that enforces changing it /var/run? From FreeBSD-side the current path is okay, is there any docs by netdata that I'm not aware of?

No, there aren't any. netdata does not enforce or check the path of the pid file. As long as the file can be created or is writable, it should be able to write it. If it fails, there should be an error log about the event (including a human readable description of the failure - i.e. permission denied, path not found, etc).

@lordcantide could you please check your `/var/log/netdata/error.log` for the reason it failed to create the pid file?

**Comment by mmokhi** [comment]
The file is being created in `/var/db/netdata/netdata.pid` as far as I tested.
It's also modifiable by `service netdata start` as a parameter in `/etc/rc.conf` as **`netdata_args`** AND **`pidfile`**

**Comment by ktsaou** [comment]
hm... then this means there is something different under pfsense...

**Comment by lordcantide** [comment]
Well this is embarrassing... I reverted to the original `/var/db/netdata/netdata.pid` file and now the service can properly reference the pid. I've attached the logs both before and after I made the PID changes, but they don't seem particularly different. To ensure I still have the backup of the error log prior to my messing with it and I see reference to 'no such process' which is what I remember getting repeatedly when attempting the service start/stop.
With an upgrade to pfsense and a minor release to netdata, time may have been a factor in troubleshooting this issue. Should I try to clean out the current netdata install and see if I can create the issue again or do you see something in the logs that I'm missing?
[error.log.backup.log](https://github.com/firehol/netdata/files/1528387/error.log.backup.log)
[error.pidchange.log](https://github.com/firehol/netdata/files/1528388/error.pidchange.log)
[error.pidoriginal.log](https://github.com/firehol/netdata/files/1528389/error.pidoriginal.log)

**Comment by ktsaou** [comment]
> Should I try to clean out the current netdata install and see if I can create the issue again or do you see something in the logs that I'm missing?

Please do. Make sure your findings are reflected to the wiki: https://github.com/firehol/netdata/wiki/Installation/_compare/22faf5a6de243b1520cad03e302a364c893161a7

**Comment by mmokhi** [comment]
@lordcantide Yes, I already checked what you attached it doesn't inform about the issue clearly (at least by the diff 😅 )
It'd be nice to ensure that we are testing in a clean-room.

**Comment by lordcantide** [comment]
I ran `pkg remove` on netdata and ensured that all netdata directories had been deleted. I then ran `pkg add` on the latest FreeBSD:11 version and found that I can no longer replicate this issue. I'm not in a place where I can rebuild from scratch at the moment so I'm going to assume pfsense and netdata are magical and self-healing 👍
When I get my real homelab together properly, I'll be sure to keep a look out for this error again. In the mean time, I've cleaned up my mess on the wiki so no one has a chance to trip over it.
